### PR TITLE
Add initial code for powervs network list

### PR DESCRIPTION
**Pull request by yussufsh**
**What this PR does / why we need it**:

Add support for listing PowerVS network.
Client code for authentication and printing results.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by yussufsh** [comment]
I have taken `utils.go` code from open PR #1071

File: cmd/capibmadm/options/options.go [hunk 1]
```diff
@@ -28,11 +32,23 @@ var GlobalOptions = &options{}
...
 	IBMCloudAPIKey    string
 	ServiceInstanceID string
+	PowerVSZone       string
+	Debug             bool
+	Output            string
+}
+
+// AddCommonFlags will add common flags to the cli.
+func AddCommonFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&GlobalOptions.Output, "output", "o", "table", "The output format of the results. Possible values: table,json,yaml")
```
Comment (Prajyot-Parab) [inline cmd/capibmadm/options/options.go:42]: I dont see yaml support so this can be removed.

File: cmd/capibmadm/cmd/powervs/network/list.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	}
+
+	for _, network := range nets.Networks {
+		listByVersion.Items = append(listByVersion.Items, NetSpec{
+			NetworkID:   *network.NetworkID,
+			Name:        *network.Name,
+			Type:        *network.Type,
+			VlanID:      *network.VlanID,
+			Jumbo:       *network.Jumbo,
+			DhcpManaged: *network.Jumbo,
```
Comment (Prajyot-Parab) [inline cmd/capibmadm/cmd/powervs/network/list.go:92]: This should be `network.Jumbo` -> `network.DhcpManaged`

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

**Comment by Prajyot-Parab** [comment]
```
prajyotparab@Prajyots-MacBook-Pro capibmadm % ./capibmadm powervs network list --service-instance-id d53da3bf-1f4a-42fa-9735-acf16b1a05cd --zone osa21 -o json
Listing PowerVS networks service-instance-id="d53da3bf-1f4a-42fa-9735-acf16b1a05cd" zone="osa21"
{
  "items": [
    {
      "id": "b0271ffe-16e3-4cdb-a3c6-3ea37832d6a5",
      "name": "capi-net-p2p",
      "type": "pub-vlan",
      "vlanID": 2024,
      "jumbo": false
    }
  ]
}%       
```
@yussufsh `DhcpManaged` should be part of items list? or above o/p is as expected?

**Comment by yussufsh** [comment]
> @yussufsh `DhcpManaged` should be part of items list? or above o/p is as expected?

I have fixed this one, initially I thought it would handle nil value as well, but it is removed from json data when the value is `false`.

**Comment by Prajyot-Parab** [comment]
```
prajyotparab@Prajyots-MacBook-Pro capibmadm % ./capibmadm powervs network list --service-instance-id d53da3bf-1f4a-42fa-9735-acf16b1a05cd --zone osa21 -o jso
n
Listing PowerVS networks service-instance-id="d53da3bf-1f4a-42fa-9735-acf16b1a05cd" zone="osa21"
{
  "items": [
    {
      "id": "dd7dd087-45ee-4147-bf8b-fb88c4130e0e",
      "name": "capi-net-p2p",
      "type": "pub-vlan",
      "vlanID": 2024,
      "jumbo": false,
      "dhcpManaged": false
    }
  ]
}%                                                                                                                                                           

prajyotparab@Prajyots-MacBook-Pro capibmadm % ./capibmadm powervs network list --service-instance-id d53da3bf-1f4a-42fa-9735-acf16b1a05cd --zone osa21      
Listing PowerVS networks service-instance-id="d53da3bf-1f4a-42fa-9735-acf16b1a05cd" zone="osa21"
NETWORK ID                             NAME           TYPE       VLAN ID   JUMBO   DHCP MANAGED
dd7dd087-45ee-4147-bf8b-fb88c4130e0e   capi-net-p2p   pub-vlan   2024      false   false
```

/lgtm

**Comment by mkumatag** [comment]
Lets wait for #1081 to merge!

**Comment by yussufsh** [comment]
> Lets wait for #1081 to merge!

@mkumatag  could you ptal?

File: cmd/capibmadm/cmd/powervs/network/list.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		return err
+	}
+
+	if options.GlobalOptions.Output == printer.PrinterTypeJSON {
+		err = pr.Print(listByVersion)
+	} else {
+		table := listByVersion.ToTable()
+		err = pr.Print(table)
+	}
```
Comment (mkumatag) [inline cmd/capibmadm/cmd/powervs/network/list.go:107]: I see a scope for optimising the Printer function itself to handle this logic, definitely not part of this PR!

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**